src/opm/parser/eclipse/EclipseState/Schedule/Well.cpp

```diff
@@ -208,6 +209,11 @@ namespace Opm {
         return m_solventFraction.update(timeStep, fraction);
     }
 
+    bool Well::setTracerProperties(size_t timeStep , const WellTracerProperties& newProperties) {
+        m_isProducer.update(timeStep , false);
```

Comment:
This is slightly schizofrenic - at an outer scope you throw an exception if the well in a question is a producer, whereas here you just update the status become an injector and go on? I would suggest only having the test in the `setTracerProperties()` - and let that throw?

Reply:
Hopefully less schizofrenic after moving the throw.